File: generator/src/test/resources/com/cromoteca/samples/endpoints/CRUDEndpoint.ts
import client from '../../../../connect-client.default.js';
import type MyLibraryType from '../../../example/library/MyLibraryType.js';
import { type EndpointRequestInit } from '@hilla/frontend';

async function doSomethingWith(t: MyLibraryType, init?: EndpointRequestInit): Promise<MyLibraryType> {
    return client.call('com.cromoteca.samples.endpoints.CRUDEndpoint', 'doSomethingWith', { t }, init);
}

async function getAll(init?: EndpointRequestInit): Promise<Array<MyLibraryType>> {
    return client.call('com.cromoteca.samples.endpoints.CRUDEndpoint', 'getAll', {}, init);
}

async function remove(t: MyLibraryType, init?: EndpointRequestInit): Promise<MyLibraryType> {
    return client.call('com.cromoteca.samples.endpoints.CRUDEndpoint', 'remove', { t }, init);
}

async function saveOrUpdate(t: MyLibraryType, init?: EndpointRequestInit): Promise<MyLibraryType> {
    return client.call('com.cromoteca.samples.endpoints.CRUDEndpoint', 'saveOrUpdate', { t }, init);
}

const CRUDEndpoint = {
    doSomethingWith,
    getAll,
    remove,
    saveOrUpdate,
};

export default CRUDEndpoint;

File: endpoints/src/main/java/com/cromoteca/samples/endpoints/CRUDEndpoint.java
package com.cromoteca.samples.endpoints;

import com.example.library.CRUD;
import com.example.library.MyLibraryType;
import com.vaadin.hilla.Endpoint;
import java.util.List;

@Endpoint
public class CRUDEndpoint implements CRUD<MyLibraryType> {
  @Override
  public List<MyLibraryType> getAll() {
    return null;
  }

  @Override
  public MyLibraryType saveOrUpdate(MyLibraryType t) {
    return t;
  }

  @Override
  public MyLibraryType remove(MyLibraryType t) {
    return t;
  }

  public MyLibraryType doSomethingWith(MyLibraryType t) {
    return t;
  }
}
